Main.py

```diff
@@ -711,32 +851,103 @@ def __init__(self, controller, selection):
+
+        # Add a button to apply the filters
+        ttk.Button(self, text='Apply Filters', command=self.apply_filters).pack(side='top', pady=5)
+
+    def apply_filters(self):
+        # Fetch values from entry fields, applying defaults if necessary
+        try:
+            min_length = int(self.min_length_entry.get()) if self.min_length_entry.get().strip() else 0
+        except ValueError:
+            min_length = 0  # Use default if input is not a valid integer
 
-        listbox = tk.Listbox(self, width=30, height=4)
-        listbox.pack(side='top', pady=5)
-        listbox.insert('end', f'{self.module.TITLE_PER_DAY} - {all_msgs / (sec_since_start / 86400):.2f}')
-        listbox.insert('end', f'{self.module.TITLE_PER_WEEK} - {all_msgs / (sec_since_start / (7 * 86400)):.2f}')
-        listbox.insert('end', f'{self.module.TITLE_PER_MONTH} - {all_msgs / (sec_since_start / (30 * 86400)):.2f}')
-        listbox.insert('end', f'{self.module.TITLE_PER_YEAR} - {all_msgs / (sec_since_start / (365 * 86400)):.2f}')
+        try:
+            max_length = int(self.max_length_entry.get()) if self.max_length_entry.get().strip() else 10000000
+        except ValueError:
+            max_length = 10000000  # Use default if input is not a valid integer
+
+        # Update values in the MasterWindow
+        self.controller.min_message_length = min_length
+        self.controller.max_message_length = max_length
+
+        # Optionally refresh or re-fetch the data based on these new settings
+        self.refresh_data_based_on_length()
+
+    def refresh_data_based_on_length(self):
+        if self.controller.current_conversation:
+            filtered_data = self.controller.get_filtered_data(self.controller.current_conversation)
+            self.update_ui(filtered_data)
+        else:
+            print("No current conversation set to apply filters.")
+
+    def update_ui(self, data):
+        self.msg_label.config(text=f'{self.module.TITLE_NUMBER_OF_MSGS}: {data["total_messages"]}')
+        self.chars_label.config(text=f'{self.module.TITLE_TOTAL_CHARS}: {data["total_chars"]}')
+        self.photos_label.config(text=f'{self.module.TITLE_NUMBER_OF_PHOTOS}: {data["total_photos"]}')
+        self.gifs_label.config(text=f'{self.module.TITLE_NUMBER_OF_GIFS}: {data["total_gifs"]}')
+        self.videos_label.config(text=f'{self.module.TITLE_NUMBER_OF_VIDEOS}: {data["total_videos"]}')
+        self.files_label.config(text=f'{self.module.TITLE_NUMBER_OF_FILES}: {data["total_files"]}')
+        self.calls_label.config(text=f'{self.module.TITLE_CALL_DURATION}: {timedelta(seconds=data["call_duration"])}')
+        self.start_date_label.config(text=f'{self.module.TITLE_START_DATE}: {datetime.fromtimestamp(data["start_date"] / 1000).strftime("%Y-%m-%d %H:%M:%S")}')
+
+        sec_since_start = int(time() - data["start_date"] / 1000)
+        avg_msgs_day = data["total_messages"] / (sec_since_start / 86400) if sec_since_start > 86400 else data["total_messages"]
```

Comment:
"Magic numbers": it's good to assign these constants (like 86400) to variables rather than repeating them.